src/core/server/http/http_service.ts

```diff
@@ -150,4 +167,28 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
     await Promise.all([...this.secondaryServers.values()].map(s => s.stop()));
     this.secondaryServers.clear();
   }
+
+  private async runNotReadyServer(config: HttpConfig) {
+    const httpServer = new HttpServer(this.log);
+    const { server } = await httpServer.setup(config);
+    this.notReadyServer = server;
```

Comment:
Any reason not to assign `this.notReadyServer` to `httpServer`? Seems like we could miss additional cleanup if any is added to `HttpServer`'s stop method.

Reply:
1. this logic is not relevant to every server. we shouldn't spin up this server for proxy server, for example.
2. we are going to cleanup `httpServer` itself
3. yes, I should add cleanup here.